Order.Management.PR/Square.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Order.Management
+{
+    class Square : Shape
+    {
+
+        public int SquarePrice = 1;
+
+        public Square(int numberOfRedSquares, int numberOfBlueSquares, int numberOfYellowSquares)
+        {
+            Name = "Square";
+            base.Price = SquarePrice;
+            AdditionalCharge = 1;
+            base.NumberOfRedShape = numberOfRedSquares;
+            base.NumberOfBlueShape = numberOfBlueSquares;
+            base.NumberOfYellowShape = numberOfYellowSquares;
+        }
+
+        public override int Total()
+        {
+            return RedSquaresTotal() + BlueSquaresTotal() + YellowSquaresTotal();
+        }
+
+        public int RedSquaresTotal()
+        {
+            return (base.NumberOfRedShape * Price);
+        }
+        public int BlueSquaresTotal()
+        {
+            return (base.NumberOfBlueShape * Price);
+        }
+        public int YellowSquaresTotal()
+        {
+            return (base.NumberOfYellowShape * Price);
+        }
+    }
+}
```

Comment:
It seems the difference between Square, Circle, and Triangle is Name, Price, and additional charge only. Therefore, we could use Shape abstract base class with properties `NumberOfRedShape`, `NumberOfYellowShape`, `NumberOfBlueShape` and function `Total`. The properties `Name`, `Price`, and `AdditionalCharge` could be defined as abstract. Also all shapes set the color shape numbers in the constructor. Therefore we could implement it in the base constructor
```
public class Shape {
    public Shape(int numberOfRedSquares, int numberOfBlueSquares, int numberOfYellowSquares)
    {
            NumberOfRedShape = numberOfRedSquares;
            NumberOfBlueShape = numberOfBlueSquares;
            NumberOfYellowShape = numberOfYellowSquares;
   }
   public int NumberOfRedShape {get; }
   ...

   public int Total() {
       return RedSquaresTotal() + BlueSquaresTotal() + YellowSquaresTotal();
    }

    private int RedSquaresTotal()
    {
        return (NumberOfRedShape * Price);
     }
     private int BlueSquaresTotal()
    {
         return (NumberOfBlueShape * Price);
     }
     private int YellowSquaresTotal()
     {
          return (NumberOfYellowShape * Price);
     }

     public abstract string Name {get; }
     ...

}

```